app/grandchallenge/codebuild/client.py

```diff
@@ -0,0 +1,85 @@
+from time import sleep
+
+import boto3
+from django.conf import settings
+from django.core import files
+from django.core.files.temp import NamedTemporaryFile
+
+from grandchallenge.algorithms.models import AlgorithmImage
+
+
+class CodeBuildClient:
+    build_id = None
+
+    def __init__(
+        self, *, project_name,
+    ):
+        self.client = boto3.client(
+            "codebuild",
+            aws_access_key_id=settings.CODEBUILD_ACCESS_KEY,
+            aws_secret_access_key=settings.CODEBUILD_SECRET_KEY,
+            region_name="eu-central-1",
+        )
+        self.s3_client = boto3.client(
+            "s3",
+            aws_access_key_id=settings.CODEBUILD_ACCESS_KEY,
+            aws_secret_access_key=settings.CODEBUILD_SECRET_KEY,
+            region_name="eu-central-1",
+        )
+        self.project_name = project_name
+        self.config = {
+            "name": project_name,
+            "source": {"type": "S3", "location": ""},
+            "artifacts": {
+                "type": "S3",
+                "location": settings.PRIVATE_S3_STORAGE_KWARGS["bucket_name"],
+            },
+            "environment": {
+                "type": "LINUX_CONTAINER",
+                "image": "docker:dind",
+                "computeType": "BUILD_GENERAL1_SMALL",
+                "privilegedMode": True,
+                "registryCredential": {
+                    "credential": settings.DOCKER_SECRET_ARN,
+                    "credentialProvider": "SECRETS_MANAGER",
+                },
+                "imagePullCredentialsType": "SERVICE_ROLE",
+            },
+            "serviceRole": settings.CODEBUILD_SERVICE_ROLE,
```

Comment:
What permissions does the service role need?

Reply:
I'll narrow down the permissions to a minimum after I finish this PR and let you know.